cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
I don't think our users will have different requirements from other Kubernetes users. Are other users okay with it not working with upstream DNS servers? 

What is the general expectation for pods? Should they be able to reach external networks (DNS or otherwise) without using the hostNetwork option?